src/libostree/ostree-repo-commit.c

```diff
@@ -2201,6 +2201,24 @@ ostree_repo_commit_transaction (OstreeRepo                  *self,
       return FALSE;
   g_clear_pointer (&self->txn.collection_refs, g_hash_table_destroy);
 
+  /* Update the summary if change-update-summary is set, because doing so was
+   * delayed for each ref change during the transaction.
+   */
+  {
+    gboolean update_summary;
+
+    if (!ot_keyfile_get_boolean_with_default (self->config, "core",
+                                              "change-update-summary", FALSE,
+                                              &update_summary, error))
+      return FALSE;
+
+    if (update_summary && !ostree_repo_regenerate_summary (self,
+                                                           NULL,
+                                                           cancellable,
+                                                           error))
+      return FALSE;
```

Comment:
At this point, it might be worth adding a `update_summary` field in `OstreeRepo`, or at least factor this out to a helper function?

Reply:
If it were a field initialized when `OstreeRepo` is initialized, there's a possibility its on-disk value is different by the time this function is run. Sure, we could make it a helper function.